torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1554,6 +1554,15 @@ int listAdd(Stack& stack) {
   push(stack, std::move(ret));
   return 0;
 }
+template <class T>
+int listInplaceAdd(Stack& stack) {
+  c10::ListPtr<T> a = c10::make_list<T>();
+  c10::ListPtr<T> b = c10::make_list<T>();
+  pop(stack, a, b);
+  a.append(b);
```

Comment:
a.append(std::move(b))

is probably what you want instead, otherwise this shouldn't catch the optimization because the local `b` variable will keep a refcount instance.